roles/node/tasks/install-avalanchego.yml

```diff
@@ -50,6 +50,10 @@
   register: avalanchego_gpg_import
   changed_when: "'imported: 1' in avalanchego_gpg_import.stderr"
 
+- name: Set avalanchego_binary_arch according to ansible_architecture
+  set_fact:
+    avalanchego_binary_arch: "{{ 'amd64' if ansible_architecture == 'x86_64' else 'arm64' if ansible_architecture == 'aarch64' else avalanchego_binary_arch }}"
```

Comment:
I think this won't work. I recently tried to overwrite a variable with `set_fact` and it has a very low priority. We should use an intermediate variable here, e.g. `binary_arch`

Reply:
See https://stackoverflow.com/a/30134080

Reply:
I did test by hardcoding the `set_fact` value of this task to `not_supported_arch` and it worked:

```
fatal: [validator04]: FAILED! => changed=false
  dest: /opt/avalanche/avalanchego/releases/avalanchego-linux-not_supported_arch-v1.10.9.tar.gz
  elapsed: 0
  msg: Request failed
  response: 'HTTP Error 404: Not Found'
  status_code: 404
  url: https://github.com/ava-labs/avalanchego/releases/download/v1.10.9/avalanchego-linux-not_supported_arch-v1.10.9.tar.gz
```

Reply:
Interesting